source/MaterialXTest/ShaderValid.cpp

```diff
@@ -50,7 +50,48 @@ namespace mx = MaterialX;
 extern void loadLibrary(const mx::FilePath& file, mx::DocumentPtr doc);
 extern void loadLibraries(const mx::StringVec& libraryNames, const mx::FilePath& searchPath, mx::DocumentPtr doc,
                           const std::set<std::string>* excludeFiles = nullptr);
-extern void createLightRig(mx::DocumentPtr doc, mx::HwLightHandler& lightHandler, mx::HwShaderGenerator& shadergen, const mx::GenOptions& options);
+
+void createLightRig(mx::DocumentPtr doc, mx::HwLightHandler& lightHandler, mx::HwShaderGenerator& shadergen, const mx::GenOptions& options,
+                    const mx::FilePath&  envIrradiancePath, const mx::FilePath& envRadiancePath)
+{
+    // Scan for lights
+    const std::string LIGHT_SHADER_TYPE("lightshader");
+    std::vector<mx::NodePtr> lights;
+    for (mx::NodePtr node : doc->getNodes())
```

Comment:
This is annoying that there is no in traversal filter. Will see about adding one.